pubtools/_quay/signer_wrapper.py

```diff
@@ -499,25 +503,62 @@ def _list_signatures(self, repository: str, tag: str) -> List[Tuple[str, str]]:
             List[Tuple[str, str]]: List of (repository, signature tag) tuples
             for existing signatures.
         """
-        full_reference = (
-            f"{self.settings['quay_host']}/"
-            + f"{self.settings['quay_namespace']}/{repository.replace('/','----')}"
-            + f":{tag}"
-        )
-        existing_signatures = run_entrypoint(
-            ("pubtools-sign", "modules", "pubtools-sign-cosign-signature-list"),
-            None,
-            [cast(str, self.config_file), full_reference],
-            {},
+        quay_client = QuayClient(
+            self.settings["dest_quay_user"],
+            self.settings["dest_quay_password"],
+            self.settings["quay_host"],
         )
-        if existing_signatures[0]:
-            return [
-                (repository, e.split(":")[-1].replace(".sig", "").replace("-", ":"))
-                for e in existing_signatures[1]
+        try:
+            ml = quay_client.get_manifest(
```

Comment:
[1] pubtools/_quay/iib_operations.py: signer.remove_signatures(outdated_manifests, _exclude=current_signatures)
[2] pubtools/_quay/push_docker.py: signer.remove_signatures(outdated_manifests, _exclude=current_signatures)
[3] pubtools/_quay/tag_docker.py: signer.remove_signatures(outdated_manifests, _exclude=current_signatures)
[4] pubtools/_quay/tag_docker.py: signer.remove_signatures(outdated_manifests, _exclude=current_signatures)
[5] pubtools/_quay/tag_docker.py: signer.remove_signatures(outdated_manifests, _exclude=current_signatures)
[6] pubtools/_quay/tag_docker.py: signer.remove_signatures(to_unsign_entries, _exclude=[])
[7] pubtools/_quay/tag_docker.py: signer.remove_signatures(to_remove_sig_entries, _exclude=[])
[8] https://github.com/release-engineering/pubtools-quay/blob/master/pubtools/_quay/image_untagger.py#L277

Removing cosign signatures in [6] is unnecessary. It's covered by [8].

To make removals in [1]-[5] right, _list_signatures needs to list digests only. If it lists a tag, the tag has been overwritten, so signatures of new manifests rather than outdated manifests would be listed. Also, [1]-[5] and [7] need similar filter as [8] to only remove signatures for `lost digests`.

If it's too complex to get the right signatures to remove, I think not removing is better than removing wrong ones.